edb/edgeql/compiler/policies.py

```diff
@@ -314,12 +311,58 @@ def try_type_rewrite(
     type_rewrites[rw_key] = filtered_set
 
 
-def compile_dml_policy(
+def compile_dml_write_policies(
```

Comment:
Think about whether there is a reasonably nice way to reduce the duplication between the read and the write side. There might not be (sometimes the cure to a little bit of duplication is worse than the disease), but think about it